worlds/stardew_valley/mods/logic/skills_logic.py

```diff
@@ -103,7 +104,13 @@ def can_earn_cooking_skill_level(self, level: int) -> StardewRule:
             return self.logic.cooking.can_cook()
 
     def can_earn_binning_skill_level(self, level: int) -> StardewRule:
-        if level >= 6:
-            return self.logic.has(Machine.recycling_machine)
-        else:
-            return True_()  # You can always earn levels 1-5 with trash cans
+        binning_rule = True_()   # Lets design this around the new craftables to spread out the logic more.
+        if level > 2:
+            binning_rule = binning_rule & self.logic.crafting.can_craft_by_name(ModMachine.trash_bin) & self.logic.crafting.can_craft_by_name(Machine.recycling_machine)
```

Comment:
Why do you care about the ability to craft thee machines specifically? Wouldn't just owning the machines be a simpler, more accurate rule?

And this would also remove the need for the `can_craft_by_name()` mehod you added above, which felt suspicious to me. `can_craft()` rules exist specifically for being able to do that crafting action, generally for craftsanity checks but also indirectly used in the rules for having access to the item.